src/ui/SettingsButton.tsx

```diff
@@ -25,8 +25,7 @@ export default function SettingsCard(
   const open = () => setIsOpen(!isOpen);
   const close = () => setIsOpen(false);
 
-  const { navigator, state, type } = props;
```

Comment:
why get rid of this?

Reply:
Typescript was throwing errors about the Type after I separate the PdfNavigator and HtmlNavigator. 

I would get `Type 'PdfNavigator | HtmlNavigator' is not assignable to type 'PdfNavigator'.` error if I destructuring the props before passing it to the PdfSettings or HtmlSettings component. I think the reason was that at the time of destructuring, the `navigator` still has the type 'PdfNavigator | HtmlNavigator' but the settings component requires a specific navigator.

Reply:
Exactly, when destructuring you lose some context typescript might have about the outer value (props in this case). Thus it becomes more difficult to refine these into one of the two types we need. It's easier in the end to keep it all together in `props`